packages/flutter_tools/lib/src/migrate/migrate_utils.dart

```diff
@@ -0,0 +1,362 @@
+      return createFromTemplates(
+        flutterBinPath,
+        name: name,
+        legacyNameParameter: true,
+        androidLanguage: androidLanguage,
+        iosLanguage: iosLanguage,
+        outputDirectory: outputDirectory,
+        platforms: platforms,
+        iterationsAllowed: iterationsAllowed--,
+      );
+    }
+    if (error.contains('Multiple output directories specified.')) {
+      if (error.contains('Try moving --platforms')) {
+        return createFromTemplates(
+          flutterBinPath,
+          name: name,
+          legacyNameParameter: legacyNameParameter,
+          androidLanguage: androidLanguage,
+          iosLanguage: iosLanguage,
+          outputDirectory: outputDirectory,
+          iterationsAllowed: iterationsAllowed--,
+        );
+      }
+    }
+    checkForErrors(result, commandDescription: cmdArgs.join(' '), silent: true);
+
+    if (legacyNameParameter) {
+      return _fileSystem.path.join(outputDirectory, name);
+    }
+    return outputDirectory;
+  }
+
+  /// Runs the git 3-way merge on three files and returns the results as a MergeResult.
+  ///
+  /// Passing the same path for base and current will perform a two-way fast forward merge.
+  Future<MergeResult> gitMergeFile({
+    required String base,
+    required String current,
+    required String target,
+    required String localPath,
+  }) async {
+    final List<String> cmdArgs = <String>['git', 'merge-file', '-p', current, base, target];
+    final RunResult result = await _processUtils.run(cmdArgs);
+    checkForErrors(result, allowedExitCodes: <int>[-1], commandDescription: cmdArgs.join(' '));
+    return StringMergeResult(result, localPath);
+  }
+
+  /// Calls `git init` on the workingDirectory.
+  Future<void> gitInit(String workingDirectory) async {
+    final List<String> cmdArgs = <String>['git', 'init'];
+    final RunResult result = await _processUtils.run(cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, commandDescription: cmdArgs.join(' '));
+  }
+
+  /// Returns true if the workingDirectory git repo has any uncommited changes.
+  Future<bool> hasUncommitedChanges(String workingDirectory) async {
+    final List<String> cmdArgs = <String>['git', 'diff', '--quiet', 'HEAD', '--', '.'];
+    // windows uses double quotes.
+    if (_platform.isWindows) {
+      String path = _fileSystem.path.join(workingDirectory, kDefaultMigrateWorkingDirectoryName);
+      // Trim off any drive labels such as 'C:'
+      if (path.contains(':')) {
+        path = path.substring(path.indexOf(':') + 1);
+      }
+      cmdArgs.add('":(exclude)$path"');
+    } else {
+      cmdArgs.add("':(exclude)${_fileSystem.path.join(workingDirectory, kDefaultMigrateWorkingDirectoryName)}'");
+    }
+    final RunResult result = await _processUtils.run(cmdArgs, workingDirectory: workingDirectory);
+    checkForErrors(result, allowedExitCodes: <int>[-1], commandDescription: cmdArgs.join(' '));
+    print(result.stdout);
```

Comment:
If this was intended, it should use the _logger (this is probably the reason stderr is leaking in the test)

Reply:
This is temporary as I work on fixing one of the tests that's being a bit stubborn in some environments.